### DISCUSSION TITLE
Git post-checkout hook alerting when ddev files have changed?

### DISCUSSION THREAD

**Discussion by deviantintegral** [ddev/ddev#6023] [Ideas]
Git lets us run a hook after a checkout operation. I realized that it would be fairly simple to create a hook that warned team members when files in `.ddev` changed. That way, they could restart ddev if it was needed.

Of course, nothing stops us from running `ddev restart` directly, but that can take time and may not always be needed.

Would something like this make sense to add when a user runs `ddev config`?

```bash
#!/bin/bash

OLD=$1
NEW=$2
BRANCH_CHANGED=$3

if [[ $BRANCH_CHANGED -eq 1 ]]; then
  git diff --quiet $OLD..$NEW -- .ddev || echo "Files in .ddev have changed. You may need to run ddev restart."
fi
```

```console
$ git checkout -
Switched to branch 'main'
Your branch is up to date with 'origin/main'.
Files in .ddev have changed. You may need to run ddev restart.
```

**Comment by davereid** [comment]
Maybe it could be added only if there's also a `.git` directory in the same as `.ddev`, this would be neat!

**Reply by rfay** [reply]
This isn't something that DDEV would do (touching your .git directory). But certainly something that makes a good tip, or an add-on or whatever.
